### PR TITLE
Added examples of practical usage to inspect

### PR DESCRIPTION
**Pull request by smorin**
Added examples of practical usage to inspect, so that people review the documentation can understand what the inspect protocol is used for.

### CODE REVIEW

File: lib/elixir/lib/inspect.ex [hunk 1]
```diff
@@ -9,6 +9,11 @@ defprotocol Inspect do
   data structure into an algebra document. This document is then
   formatted, either in pretty printing format or a regular one.
 
+  `Inspect` protocol basically returns a string for the data
+  structure it's given. The string returned by the `Inspect` 
+  protocol, generated using both the options from `Inspect.Opts`
+  and the internal algebra representation of the datastructure.
```
Comment (fertapric) [inline lib/elixir/lib/inspect.ex]: s/datastructure/data structure

File: lib/elixir/lib/inspect.ex [hunk 2]
```diff
@@ -9,6 +9,11 @@ defprotocol Inspect do
   data structure into an algebra document. This document is then
   formatted, either in pretty printing format or a regular one.
 
+  The `Inspect` protocol basically returns a string for the data
```
[the remark this answers is not in the document]
Comment (smorin) [reply lib/elixir/lib/inspect.ex]: Thanks for the clarification, but there need to be a bit on how it's used as well which what I was trying to explain.  It's hard for people new to inspect to know how it's used.

File: lib/elixir/lib/inspect.ex [hunk 3]
```diff
@@ -40,6 +45,27 @@ defprotocol Inspect do
   an implementation of inspect may simply return a string,
   although that will devoid it of any pretty-printing.
 
+
+      iex> Kernel.inspect(23)
```
[the remark this answers is not in the document]
Comment (smorin) [reply lib/elixir/lib/inspect.ex]: I think that's a good idea.

**Comment by josevalim** [comment]
@smorin great! I use this opportunity to change the first paragraph, which should always be a short summary but it wasn't.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**